### PR TITLE
Sidebar Menus

### PR DESCRIPTION
**Pull request by marjohn56**
Sidebar Collapsible Menus

Requires changes to existing themes - seperate PR's issued.

### CODE REVIEW

File: src/opnsense/mvc/app/views/layouts/default.volt [hunk 1]
```diff
@@ -164,9 +164,10 @@
             });
         </script>
 
-        <!-- JQuery Tokenize2 (https://zellerda.github.io/Tokenize2/) -->
-        <script src="/ui/js/tokenize2.min.js"></script>
-        <link rel="stylesheet" type="text/css" href="{{theme_file_or_default('/css/tokenize2.css', theme_name)}}" rel="stylesheet" />
+
+        <!-- JQuery Tokenize (http://zellerda.com/projects/tokenize) -->
```
Comment (fichtner) [inline src/opnsense/mvc/app/views/layouts/default.volt]: Merge issue, reverts current state from master.

File: src/www/fbegin.inc [hunk 2]
```diff
@@ -62,6 +62,7 @@ if($need_alert_display == true) {
           <span class="icon-bar"></span>
         </button>
       </div>
+	  <button class="toggle-sidebar glyphicon glyphicon-chevron-left"></button>
```
Comment (fichtner) [inline src/www/fbegin.inc]: I cleaned up all glyphicons recently :/ https://github.com/opnsense/core/issues/2440

File: src/www/fbegin.inc [hunk 3]
```diff
@@ -87,8 +88,6 @@ if($need_alert_display == true) {
                     <div class="panel list-group" style="border:0px">
 <?php
                           foreach($menuSystem as $topMenuItem): ?>
-<?php
```
Comment (fichtner) [inline src/www/fbegin.inc]: Another merge issue.

File: src/www/fbegin.inc [hunk 4]
```diff
@@ -182,12 +163,12 @@ if($need_alert_display == true) {
                 }
               } ?>
               <?php if (!empty($main_buttons)): foreach ($main_buttons as $button): ?>
-                <a href="<?=$button['href'];?>" class="btn btn-primary"><i class="fa fa-plus-circle fa-fw"></i> <?= $button['label'] ?></a>
+                <a href="<?=$button['href'];?>" class="btn btn-primary"><span class="glyphicon glyphicon-plus-sign __iconspacer"></span><?=$button['label'];?></a>
```
Comment (fichtner) [inline src/www/fbegin.inc]: merge issue

File: src/www/fbegin.inc [hunk 5]
```diff
@@ -182,12 +163,12 @@ if($need_alert_display == true) {
...
               <?php endforeach; endif; ?>
 
               <?php if (isset($widgetCollection)): ?>
                 <a href="#" id="updatepref" style="display:none" onclick="return updatePref();" class="btn btn-primary"><?=gettext("Save Settings");?></a>
-                <button id="add_widget_btn" type="button" class="btn btn-default" data-toggle="modal" data-target="#modal_widgets"><i class="fa fa-plus-circle fa-fw"></i> <?= gettext('Add widget') ?></button>
+                <button id="add_widget_btn" type="button" class="btn btn-default" data-toggle="modal" data-target="#modal_widgets"><span class="glyphicon glyphicon-plus-sign __iconspacer"></span><?= gettext('Add widget') ?></button>
```
Comment (fichtner) [inline src/www/fbegin.inc]: merge issue

**Comment by AdSchellevis** [comment]
@marjohn56 is this backwards compatible with older themes? we have quite some different themes in the wild, if we can make sure older theme's keep working that would help getting this in.

**Comment by fichtner** [comment]
@AdSchellevis handing this to you as you have better understanding of the impact

**Comment by AdSchellevis** [comment]
@fichtner sure, no problem

**Comment by marjohn56** [comment]
According to René changes are required to themes. He has modified Tukan and Cicada, and tested modified Opnsense and Rebellion.

I have queried the first comment you made with René. I think first thing to do here is to make sure René is up to date with the current source.

**Comment by opnsenseuser** [comment]
I took the default.volt and the fbegin.inc from the latest Version of opnsense an put the javascode in there.

besides, fabian wanted to check the code in advance?

**Comment by fichtner** [comment]
Code diverges, it happens! No worries. We just need to make sure we only pull in the relevant changes and not undo other people's work in the process. We'll manage of course. :)

**Comment by opnsenseuser** [comment]
Only copy and paste the <button> Row and the Part from  to  into the actually default and fbegin file. Thats all

**Comment by marjohn56** [comment]
René can you do what is needed and send it to me? I'll squash the commits to correct.

**Comment by opnsenseuser** [comment]
On your way in 2 Minuten

**Comment by marjohn56** [comment]
Updated... let's see how that is now.

**Comment by fichtner** [comment]
Much better, thanks @marjohn56 and @opnsenseuser -- I'll take care of the glyphicon conversion post-merge.

**Comment by opnsenseuser** [comment]
that´s a clean merge. great! :-)

**Comment by opnsenseuser** [comment]
**@fichtner** do I have to rewrite the javascript and the main.css? because apparently all glyphicon classes get away? and in javascript i used the glyphicon classes. also in the main.css. If so, I would have to change the source code in the main.css and in the javascript for the sidebar to work.

**Comment by fichtner** [comment]
@opnsenseuser I'll take care of it after merge. It's not a theme change, just symbols in static core pages. Your themes are not affected by this and will still be ok. :)

**Comment by opnsenseuser** [comment]
@fichtner let me know if I can help you! :-)

**Comment by AdSchellevis** [comment]
@marjohn56 I've refactored the code to deduplicate it, added your usual copyright, feel free to change it if it should be different. 

The new version should also detect if the stylesheet is actually loaded, otherwise the functionality will be ignored.

**Comment by marjohn56** [comment]
Cool...

**Comment by opnsenseuser** [comment]
I‘m on holiday for 2 weeks. 
thanks for your Support and the professional work.

Regards
Rene


Gesendet von myMail für iOS


Dienstag, 17. Juli 2018, 22:03 +0200 von notifications@github.com  <notifications@github.com>:

**Comment by marjohn56** [comment]
Nice work René.

When you get back, email me before you do any further work on your themes or Sidebar,

**Comment by fichtner** [comment]
I did two, the rest is up for grabs and / or discussion, even extending the TODO list.

**Comment by marjohn56** [comment]
OK, I'll  do the theme update.

**Comment by marjohn56** [comment]
Try #2555

**Pull request merged**